### PR TITLE
added more detail for this bug

### PR DESCRIPTION
**Pull request by Maoni0**
@rpetrusha @vivmishra PTAL

### CODE REVIEW

File: releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md [hunk 1]
```diff
@@ -7,9 +7,20 @@ intermittent crashes that did not occur when running on other .NET Framework ver
 
 ## Cause
 
-.NET Framework 4.7.1 includes changes that improve background GC performance.
-Because of these changes, a rare combination of factors can cause the background GC
-to reclaim a large object while the program is still using it.
+.NET Framework 4.7.1 includes changes that improve background GC performance. It allows large object allocations during part of the BGC sweep phase and the large objects allocated during this time must be marked correctly in the allocator or it will be erroneously reclaimed. A GC can be triggered during the large object allocator code path and if the following happens this bug will be triggered:
+
+- this is a BGC
```
Comment (rpetrusha) [inline releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md]: this --> This
period after BGC

File: releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md [hunk 2]
```diff
@@ -7,9 +7,20 @@ intermittent crashes that did not occur when running on other .NET Framework ver
...
 ## Cause
 
-.NET Framework 4.7.1 includes changes that improve background GC performance.
-Because of these changes, a rare combination of factors can cause the background GC
-to reclaim a large object while the program is still using it.
+.NET Framework 4.7.1 includes changes that improve background GC performance. It allows large object allocations during part of the BGC sweep phase and the large objects allocated during this time must be marked correctly in the allocator or it will be erroneously reclaimed. A GC can be triggered during the large object allocator code path and if the following happens this bug will be triggered:
+
+- this is a BGC 
+
+- the range for BGC changed from the previous range (ie, a new segment is allocated outside of the previous range)
```
Comment (rpetrusha) [inline releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md]: the --> The (each list item should begin with a capital letter)
end in period here and for each list item

File: releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md [hunk 3]
```diff
@@ -7,9 +7,20 @@ intermittent crashes that did not occur when running on other .NET Framework ver
...
+
+- the new large object is in the region that didn't exist in the previous range
+
+- the new large object did its marking during the BGC sweep phase
+
+which makes this bug very rare.
+
+Since GC doesn't shrink the range, it means this bug usually is only seen very early on during a process or when you suddenly allocate more large objects than before which requires acquiring a new segment that's outside the heap range.
```
Comment (rpetrusha) [inline releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md]: comma before "or when"
comma after "than before"

File: releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md [hunk 4]
```diff
@@ -26,7 +37,7 @@ The most practical workaround is to disable the background GC by setting the ena
   <runtime> section of your application configuration file to false.
     
 The only other workaround is to change the application to allocate and free large objects
-at a much lower rate.
+at a much lower rate. However, please note that this workaround do not provide a guarantee - it just makes the chance of this bug happening smaller.
```
Comment (rpetrusha) [inline releases/net471/KnownIssues/470006-GC Crashes with high rate of large object allocation.md]: - "please" can be deleted
do not --> doesn't
guarantee - it --> guarantee; it

**Comment by Maoni0** [comment]
@rpetrusha when you checked in your commit there was a typo:

(https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtimea/gcconcurrent-element) 

runtimea should be runtime. I was just referring to this and tried to click on the url and it went to a page that doesn't exist. could you fix it please? thanks!

**Comment by rpetrusha** [comment]
Certainly I'll fix it, @Maoni0.

**Pull request merged**